raiden/transfer/log.py

```diff
@@ -83,6 +84,12 @@ def __init__(self, database_path):
             'FOREIGN KEY(statechange_id) REFERENCES state_changes(id)'
             ')'
         )
+        cursor.execute(
+            'CREATE TABLE IF NOT EXISTS state_events ('
+            'identifier integer primary key, source_statechange_id integer NOT NULL, data binary, '
+            'FOREIGN KEY(source_statechange_id) REFERENCES state_changes(id)'
```

Comment:
We don't specifically need this. 

I put it only to have SQL verify for us that each state event we insert into the database has a corresponding state change in the WAL since that should be a requirement. We can't have an event without the corresponding state change that generated it.

So far we save all the state changes in the WAL and thus I thought it proper to assert that an event has a corresponding change.

When/If we ever decide to save only specific state changes to the WAL then this can be removed.